### PR TITLE
Cleanup of old, now unused code for v4

### LINKED ISSUE #722: Cleanup post-add-browserify
:fire: to take place after #699:, to be removed once verified not actually needed:
- [x] concat task, nojquery/noleaflet no long needed (already done in 6ae2335b3adccaf96ceb5155033779fc657b9262)
- [x] `index.js` old rails manifest?
- [x] `src-browserify/loader.js`
- [x] `geo/ui/switcher*`
- [x] `test/lib`, `sinon` no longer used in new specs
- [x] Rename `_Promise` to `Promise`
- [x] Review files/classes that declara plain Backbone.Model and Backbone.Collection (eg: [geo/geometries.js](https://github.com/CartoDB/cartodb.js/blob/v4-browserify-standard2/src-browserify/geo/geometries.js))
- [x] `package.json` browser/browserify-shim definitions, all used?
- [x] `Makefile`, really necessary nowadays? 
- [x] `scripts` dir, e.g. `scripts/compress`, only be used for cartodbeditor; can be removed/moved?
- [x] `grunt/`, e.g. templates used in core concat, no longer needed (e.g. underscore_no_conflict, version_header etc.).
- [x] `vendor/reqwest` (unused since not supporting core or nojquery bundles anymore)
- [x] `vendor/mod/torque.uncompressed`, now required through torque.js npm bundle?

Out-of-scope/postponed:
- ~~rm `test/demos/`~~ :point_right: #715
- ~~`src/` + `test/spec` (but not `src-browserify`)~~ :point_right: #758
- CartoDB editor also will need to be updated at some point :point_right: cartodb/cartodb#6068

### PR DESCRIPTION
**Pull request by viddo**
Resolves #722 (part of #699)
See the issue and/or commits list for scope

@xavijam can you review?
 cc @alonsogarciapablo

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,91 +0,0 @@
-# Old makefile
```
Comment (xavijam) [inline Makefile]: Thanks for all your service, always in my heart, RIP makefile.

File: src-browserify/api/create-layer.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ var Loader = require('../core/loader');
 var Map = require('../geo/map');
 var Vis = require('../vis/vis');
 var Layers = require('../vis/vis/layers');
-var _Promise = require('./_promise');
+var Promise = require('./promise');
```
Comment (xavijam) [inline src-browserify/api/create-layer.js]: https://www.youtube.com/watch?v=2QOCSZcTWWc

File: vendor/reqwest.min.js [hunk 3]
```diff
@@ -1,522 +0,0 @@
-/*!
```
Comment (xavijam) [inline vendor/reqwest.min.js]: Oh! Reqwest out too? :+1:

**Comment by xavijam** [comment]
Just out of curiosity, we are removing underscore lib from vendor folder, is it ok right?

**Comment by xavijam** [comment]
Just checked that some vendor libs have been replaced by the npm lib, so :+1:.
I'd like to have a one minute of silence in name of the old Makefile.

**Comment by viddo** [comment]
> Just out of curiosity, we are removing underscore lib from vendor folder, it is ok, isn't it?
> …
> Just checked that some vendor libs have been replaced by the npm lib, so :+1:

yes, :fire: on purpose :)

File: src-browserify/api/create-layer.js (hunk 2 above)
Comment (Kartones) [inline src-browserify/api/create-layer.js]: :_)  nice choice!

File: vendor/reqwest.min.js (hunk 3 above)
Comment (viddo) [inline vendor/reqwest.min.js]: yes, since we decided to :fire: nojquery/noleaflet it could be removed, see 6c998a9a310dfc669e2dd6354a5aa26806eb55cd

Comment (xavijam) [inline vendor/reqwest.min.js]: :star:

**Pull request merged**